test/integration/authorization_test.go

```diff
@@ -622,7 +622,7 @@ func TestAuthorizationResourceAccessReview(t *testing.T) {
 				Users:           sets.NewString("edgar"),
 				Groups:          sets.NewString(),
 				Namespace:       "mallet-project",
-				EvaluationError: `[clusterrole.rbac.authorization.k8s.io "admin" not found, clusterrole.rbac.authorization.k8s.io "admin" not found]`,
+				EvaluationError: `[clusterrole.rbac.authorization.k8s.io "admin" not found, clusterrole.rbac.authorization.k8s.io "admin" not found, clusterrole.rbac.authorization.k8s.io "admin" not found]`,
```

Comment:
@enj i don't really understand why my changes in this PR would necessitate this change... can you enlighten me on what is going on here? (in general it seems weird that we'd return a string containing the same error message repeated N times w/ no additional context)

Reply:
checking who can do a thing involves checking all applicable role bindings and the roles they refer to. any bindings that reference an unresolveable role result in a lookup error. all the lookup errors are aggregated here: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac/subject_locator.go#L79

I'm not sure if we should dedupe the aggregated messages or add more info to distinguish which binding each lookup error came from (which could leak info).

In any case, you added a new clusterrolebinding which in this test becomes invalid (since the test intentionally deletes the admin role). This test might be better served creating a single-use role and a couple referencing rolebindings.

Reply:
thanks @liggitt, makes sense